### PR TITLE
which command implementation

### PR DESCRIPTION
**Pull request by nepodmitljivi**
## Description
- `yvm which` command searches users PATH for installed version

### DevQA Prep

- Please `npm install`, this PR uses a new package

### DevQA Steps
- run `yvm which`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -57,6 +57,14 @@ module.exports = function dispatch(args) {
             exec(version, extraArgs)
         }))
 
+    argParser
+        .command('which')
```
Comment (umar-tophat) [inline src/index.js]: Can you add description here?

Comment (nepodmitljivi) [reply src/index.js]: done

**Comment by umar-tophat** [comment]
If I don't have a `.yvmrc` file in the current directory, it shouldn't show this message: `your RC and PATH versions don't match :(`.

File: src/commands/which.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        const matchedPath = matchedString[0]
+        const matchedVersion = matchedPath.toString().match(versionRegex)
+        log(`matched yvm version: ${matchedVersion} in PATH ${matchedPath}`)
+
+        const pathVersion = matchedVersion.toString().replace(/v/g, '')
+        const rcVersion = getRcFileVersion()
+        if (pathVersion === rcVersion) {
+            log(`your RC version matches your PATH version, good job!`)
+        } else {
+            log(`your RC and PATH versions don't match :(`)
```
Comment (umar-tophat) [inline src/commands/which.js]: maybe add `process.exit(1)` to indicate that the two don't match.

**Comment by jakebolam** [comment]
QA, seeing this:
```
➜  yvm git:(yvm-which) yvm list
Checking for installed yarn versions...
Installed yarn versions:
  - v1.7.0

➜  yvm git:(yvm-which) yvm which
Checking yarn version
You don't have yvm version installed
```

File: src/commands/which.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const { getRcFileVersion } = require('../util/version')
+const log = require('../common/log')
+
+const whichCommand = () => {
+    const regex = /(\/Users)(\/)((?:[a-z][a-z0-9_]*))(\/)(\.yvm)(\/versions)(\/v\d+\.?\d+\.?\d+)(\/bin)/gm
```
Comment (jakebolam) [inline src/commands/which.js]: Anyway we can make this a little more readable?
We probably shouldn't depend on a user install .yvm in the `Users` folder

Comment (jakebolam) [reply src/commands/which.js]: I don't think this regex will find it.
yvm install.sh, put it under
`/usr/local/bin/yvm`
which is included in my path as . `/usr/local/bin/`

Comment (jakebolam) [reply src/commands/which.js]:
```
$PATH
zsh: no such file or directory: /Users/jakebolam/.nvm/versions/node/v6.11.5/bin:/Users/jakebolam/.yarn/bin:/Users/jakebolam/.config/yarn/global/node_modules/.bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

File: src/commands/which.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                }
+            }
+            foundVersion = true
+        }
+        return 0
+    })
+
+    if (!foundVersion) {
+        log(`You don't have yvm version installed`)
+        return 2
```
Comment (francoiscampbell) [inline src/commands/which.js]: i don't see these return values used anywhere?

**Comment by umar-tophat** [comment]
## ✅ QA

- `yvm which` shows correct message and exit code when no yarn version installed and when yarn version matches `.yvmrc`

**Pull request merged**